### ISSUE TITLE
Independent  vector-Jacobian product for multiple arguments

### ISSUE THREAD

**Issue by chaoming0625** [google/jax#8591]
### Discussed in https://github.com/google/jax/discussions/8571

<div type='discussions-op-text'>

<sup>Originally posted by **chaoming0625** November 17, 2021</sup>
In the official documentation, JAX supports ``vjp`` for multiple arguments:

```python
>>> import jax
>>>
>>> def f(x, y): return jax.numpy.sin(x), jax.numpy.cos(y)
>>> primals, f_vjp = jax.vjp(f, 0.5, 1.0)
>>> xbar, ybar = f_vjp((-0.7, 0.3))
>>> print(xbar)
-0.61430776
>>> print(ybar)
-0.2524413
```

Once the return values are using both arguments:

```python
import jax.numpy as jnp
from jax import vjp

def vgrad(f, xs):
  ys, vjp_fn = vjp(f, *xs)
  return vjp_fn(tuple([jnp.ones(y.shape) for y in ys]))


def f(x, y):
  dx = x ** 2 + y ** 2 + 10
  dy = x ** 3 + y ** 3 - 10
  return dx, dy

pprint(vgrad(f, [0.1, 0.2]))

# (0.23, 0.52)
```

However, we want to get the derivative for each argument independently.  That is to say, the except derivatives are:

```python

def expect(x, y):
  return 2 * x, 3 * y ** 2

expect(0.1, 0.2)

# (0.2, 0.12)
```
 
Can anyone tell me how to make the excepted gradients? Thanks!

**Comment by hawkinsp** [comment]
Closing because you have also opened a discussion (https://github.com/google/jax/discussions/8571). Please only open one.
